From fa054fa44f2f95acb03ba07df001ad722b492a47 Mon Sep 17 00:00:00 2001
From: Andrew Youll <andrewyoull86@gmail.com>
Date: Sun, 27 Sep 2026 11:51:06 +0100
Subject: [PATCH 09/11] swscale arm: YUV->RGB NEON stores don't assume a
 16-byte aligned destination

The converters wrote with vst4.8 {..}, [dst,:128], which faults when the
caller's buffer isn't 16-byte aligned (seen writing straight into a
4-byte aligned EGL surface / sprite).
---
 libswscale/arm/yuv2rgb_neon.S | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/libswscale/arm/yuv2rgb_neon.S b/libswscale/arm/yuv2rgb_neon.S
index 10950e7..9684f6c 100644
--- a/libswscale/arm/yuv2rgb_neon.S
+++ b/libswscale/arm/yuv2rgb_neon.S
@@ -75,8 +75,10 @@
     vzip.8              d7, d11                                        @ d7 = G1G2G3G4G5G6G7G8 d11 = G9G10G11G12G13G14G15G16
     vzip.8              d8, d12                                        @ d8 = B1B2B3B4B5B6B7B8 d12 = B9B10B11B12B13B14B15B16
     vzip.8              d9, d13                                        @ d9 = A1A2A3A4A5A6A7A8 d13 = A9A10A11A12A13A14A15A16
-    vst4.8              {q3, q4}, [\dst,:128]!
-    vst4.8              {q5, q6}, [\dst,:128]!
+    @ No :128 on dst: it can be any caller's buffer (a sprite or a locked
+    @ EGL surface on RISC OS is only 4-byte aligned), and .8 needs none.
+    vst4.8              {q3, q4}, [\dst]!
+    vst4.8              {q5, q6}, [\dst]!
 .endm
 
 .macro process_1l_internal dst src ofmt
